### ISSUE TITLE
Search for prev/next sibling returns Collection

### ISSUE THREAD

**Issue by Karmalakas** [getgrav/grav#3396]
_/system/src/Grav/Common/Page/Collection.php::adjacentSibling()_ may return back the whole _Collection_ if previous or next sibling is not found. All other places in such case expect to return `false`. _system/src/Grav/Common/Page/Interfaces/PageCollectionInterface.php::adjacentSibling()_ defines return types `PageInterface|PageCollectionInterface|false`, but `PageCollectionInterface` should not be there as no other method expects to return it. Also naming suggests a `PageInterface` if found or empty value (`null|false`) to be returned if not found, but not a collection

**Comment by mahagr** [comment]
Yup. I have no idea why it does that, but I did add `PageCollectionInterface` into the return values because it did that for some crazy reason.

CC @rhukster

**Comment by Karmalakas** [comment]
FYI, I'm already creating a PR for this 😉

**Comment by mahagr** [comment]
I don't think it will be accepted as I'm pretty sure there was a reason why I didn't fix that.

**Comment by Karmalakas** [comment]
Still created 🙂 Maybe it'll be a jump-start for Grav team to investigate this

**Comment by mahagr** [comment]
I checked Flex Pages collections and it already has this behavior. Though it's not really being used, but... I must have been thinking about changing the behavior already.

**Comment by Karmalakas** [comment]
Yes, elsewhere it's fine. That's why I'm leaning towards fixing page collection too.

**Comment by praserx** [comment]
Hi, any progress here? I am facing the same issue as @Karmalakas and it is quite annoying. Is there any correct (simple or straightforward) way to get page siblings?

I saw similar behavior with `{{ page.parent().children() }}` where I am getting also collection (I mean whole pages collection). I expect to get collection of pages of specific scope. Same, if I call `{{ page.parent().collenction() }}` I expect specific collection of pages, not whole collection.

For me, it looks like it is totally broken, or I am wrong? Or I don't get something important?
